src/golang/lib/job/databricks.go

```diff
@@ -0,0 +1,189 @@
+package job
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aqueducthq/aqueduct/lib/collections/shared"
+	databricks_lib "github.com/aqueducthq/aqueduct/lib/databricks"
+	databricks_sdk "github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"github.com/dropbox/godropbox/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	databricksFunctionScript = "aqscript.py"
+	databricksParamScript    = "paramScript.py"
+	databricksMetricScript   = "metricScript.py"
+	databricksDataScript     = "dataScript.py"
+)
+
+type DatabricksJobManager struct {
+	databricksClient *databricks_sdk.WorkspaceClient
+	conf             *DatabricksJobManagerConfig
+	runMap           map[string]int64
+}
+
+func NewDatabricksJobManager(conf *DatabricksJobManagerConfig) (*DatabricksJobManager, error) {
+	databricksClient, err := databricks_lib.NewWorkspaceClient(
+		conf.WorkspaceUrl,
+		conf.AccessToken,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to create Databricks Workspace Client.")
+	}
+
+	return &DatabricksJobManager{
+		databricksClient: databricksClient,
+		conf:             conf,
+		runMap:           map[string]int64{},
+	}, nil
+}
+
+func (j *DatabricksJobManager) mapJobTypeToFile(spec Spec) (string, string, error) {
+	if spec.Type() == FunctionJobType {
+		functionSpec, ok := spec.(*FunctionSpec)
+		if !ok {
+			return "", "", ErrInvalidJobSpec
+		}
+
+		functionSpec.FunctionExtractPath = defaultFunctionExtractPath
+		storageConfig, err := spec.GetStorageConfig()
+		if err != nil {
+			return "", "", errors.Wrap(err, "Spec unexpectedly has no storage config.")
+		}
+		storageConfig.S3Config.AWSAccessKeyID = j.conf.AwsAccessKeyId
+		storageConfig.S3Config.AWSSecretAccessKey = j.conf.AwsSecretAccessKey
+
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricksFunctionScript, specStr, nil
+
+	} else if spec.Type() == ParamJobType {
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricksParamScript, specStr, nil
+
+	} else if spec.Type() == AuthenticateJobType ||
+		spec.Type() == LoadJobType ||
+		spec.Type() == ExtractJobType ||
+		spec.Type() == LoadTableJobType ||
+		spec.Type() == DeleteSavedObjectsJobType ||
+		spec.Type() == DiscoverJobType {
```

Comment:
Are these expected to actually work? The issue is just data scale right.

Reply:
These do work, they just don't read directly into spark so we don't solve the scale issue.